File: backend/internal/models/type.go
package models

import "time"

// Type is a resource-type-agnostic, team-customizable category (for example an
// artifact's "Work reports" / "Bug report"). System defaults are global rows
// (TeamID empty, IsSystem true) visible to every team; custom types belong to a
// single team. Uniqueness is on (team_id, resource_type, slug).
type Type struct {
	ID           string    `json:"id" db:"id"`
	TeamID       string    `json:"team_id,omitempty" db:"team_id"`
	ResourceType string    `json:"resource_type" db:"resource_type"`
	Slug         string    `json:"slug" db:"slug"`
	Name         string    `json:"name" db:"name"`
	IsSystem     bool      `json:"is_system" db:"is_system"`
	CreatedBy    string    `json:"created_by,omitempty" db:"created_by"`
	CreatedAt    time.Time `json:"created_at" db:"created_at"`
	UpdatedAt    time.Time `json:"updated_at" db:"updated_at"`
}

File: frontend/src/types/type.ts
// Resource-type-agnostic, team-customizable categories ("types"). System
// defaults are global and read-only (is_system true, no team_id); custom types
// belong to a team. Mirrors the backend `types` table (#1846).
export interface Type {
  id: string
  team_id?: string
  resource_type: string
  slug: string
  name: string
  is_system: boolean
  created_at: string
}

export interface CreateTypeRequest {
  resource_type: string
  slug: string
  name: string
}

export interface TypeListResponse {
  types: Type[]
  total_count: number
}
